src/components/menu/menu-popover.tsx

```diff
@@ -11,8 +11,8 @@ export const MenuPopover: FC = ({ children }) => {
 
   if (isOpen) {
     const onClick = (e) => {
-      // close menu if button or any other element inside e.g icon is clicked
-      const button = e.target.closest('button')
+      // close menu if button or any other element inside e.g icon button is clicked
+      const button = e.target.closest('button, [role="button"]')
```

Comment:
to solve possible issue when consumer use `<div role="button">" for list item

Reply:
i think the menu tickbox version could require the popover to persist so the user can make selections

Reply:
just found the feature that might allow checkbox or radio to close the menu here https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-editor

but as long as consumer are using it properly e.g ```<div role="menuitemradio">This is a radio div</div>``` I think it will be okay, I will make an example to include the role in next pr for radio and checkbox 👍

Reply:
**note:** In reading https://www.w3.org/WAI/ARIA/apg/patterns/menubar/, the guidance for checkbox menu items states, when discussing keyboard interaction for the Space key:

> (Optional): When focus is on a menuitemcheckbox, changes the state without closing the menu.

This suggests that we may want that behaviour _by default_ for checkbox menu items, which increases the likelihood that we'll need to decide how to facilitate that behaviour.

Reply:
> todo: @nurm717123 I think Adrian raises a good point. Can you please ask Design if they want that behaviour to be available?

ok

> This suggests that we may want that behaviour by default for checkbox menu items

yes, this is will be default as current behaviour